### PR TITLE
Jsonforms for form component

### LINKED ISSUE #179: root .content.xml references admin dependencies
**Describe the bug**

See `siteJS` and `siteCSS`

```xml
<?xml version="1.0" encoding="UTF-8"?>
<jcr:root
  xmlns:sling="http://sling.apache.org/jcr/sling/1.0"
  xmlns:jcr="http://www.jcp.org/jcr/1.0"
  jcr:primaryType="per:Page">

  <jcr:content
    jcr:primaryType="per:PageContent"
    sling:resourceType="themecleanflex/components/page"
    jcr:title="themecleanflex root template"
    siteJS="[/etc/felibs/admin/dependencies/vfg.js,/content/themecleanflex/pages/js/formsapp.js]"
    siteCSS="[/etc/felibs/admin/dependencies/vfg.css,/content/themecleanflex/pages/css/commons.css,/content/themecleanflex/pages/css/palettes/default.css,/content/themecleanflex/pages/css/palette.css,/etc/felibs/themecleanflex.css,/content/themecleanflex/pages/css/custom.css,https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;700;800&amp;display=swap]"
    prefetchDNS="[https://www.youtube.com,https://s.ytimg.com,https://www.google.com,https://fonts.gstatic.com,https://www.youtube-nocookie.com]"/>

    <footer/>
    <blank/>

</jcr:root>
```

### PR DESCRIPTION
**Pull request by flx-sta**
## Known "Issues"
- Currently a workaround is used to ensure correct re-rendering of the form (see #182)
- All script files had to be imported inside [`renderer.html (L. 11++)`](https://github.com/headwirecom/themeclean-flex/pull/181/files#diff-08d13407b8259232f1a44295f6153fc944f2e2608e1f9a427dd8e7412f7e9bd8R11) because otherwise it wasn't possible with the current architecture to ensure the correct import order

Closes #179

### CODE REVIEW

**Comment by reusr1** [comment]
Screenshot from 2021-03-26 19-33-41

seems like once I install this PR the drop targets in the editor disappear

**Comment by reusr1** [comment]
Screenshot from 2021-03-26 19-36-50

I took the basic example from the jsonforms.io homepage at https://jsonforms.io/ - seems it doesnt look that great (like the group is not below)

**Comment by flx-sta** [comment]
> [image]
> 
> seems like once I install this PR the drop targets in the editor disappear

@reusr1 
seems like this is a general issue in peregrine-cms/develop-sling12
I've created a PR for it: https://github.com/headwirecom/peregrine-cms/pull/826

> [image]
> 
> I took the basic example from the jsonforms.io homepage at https://jsonforms.io/ - seems it doesnt look that great (like the group is not below)

@reusr1 
Yeah apparently I had a wrong version of the json-forms core causing `VerticalLayout` and `HorizontalLayout` to be ignored

**Comment by Felix-Puetz** [comment]
> [image]
> 
> I took the basic example from the jsonforms.io homepage at https://jsonforms.io/ - seems it doesnt look that great (like the group is not below)

Yeah apparently I had a wrong version of the json-forms core causing `VerticalLayout` and `HorizontalLayout` to be ignored

**Comment by cmrockwell** [comment]
I’m finding an issue in jsonforms-for-form-component the container for dropping components is missing.

The image below shows a new page created from current develop-sling12 (left) and jsonforms-for-form-component (right). In both cases the template having header and footer were used.

image

File: ui.apps/src/main/content/jcr_root/apps/themecleanflex/components/page/renderer.html [hunk 1]
```diff
@@ -8,6 +8,12 @@
 
 <script src="/etc/felibs/themecleanflex/dependencies/intersection-observer.js"></script>
 <script src="/etc/felibs/themecleanflex/dependencies/v-lazy-image.min.js"></script>
+<script src="/etc/felibs/themecleanflex/dependencies/sha256.js"></script>
+<script src="/etc/felibs/themecleanflex/dependencies/vue-2.6.js"></script>
```
Comment (reusr1) [inline ui.apps/src/main/content/jcr_root/apps/themecleanflex/components/page/renderer.html]: this is an issue, we really should instead upgrade the pagerendervue version in the main peregrine project

Comment (flx-sta) [reply ui.apps/src/main/content/jcr_root/apps/themecleanflex/components/page/renderer.html]: I'm fine with it but I still think it's a huge flaw in the architecture. That the theme relies on the main project for rendering. but maybe I just mis understanding the whole "headless" concept

Comment (reusr1) [reply ui.apps/src/main/content/jcr_root/apps/themecleanflex/components/page/renderer.html]: @Felix-Puetz to clear this up a bit: themeclean-flex relies on peregrine-cms/pagerenderer/vue for the base functionality of how to render a page - a theme can be fully independent but then it needs to implement the functionality provided by peregrine-cms/pagerender/vue - the admin console also builds on peregrine-cms/pagerenderer/vue - themecleanflex declares its dependency on pagerendervue at https://github.com/headwirecom/themeclean-flex/blob/develop/ui.apps/src/main/content/jcr_root/content/themecleanflex/.content.xml (see the sourceSite property)

**Comment by flx-sta** [comment]
@reusr1 I've added a solution for the dependency order loading (for now).
unfortunately it is not the pretties nor the most convinient but there is no other right now without having to add a  new concept of loading `siteJS` and renderer.html stuff

**Review by reusr1 — APPROVED** [review]

**Review by reggie7 — APPROVED** [review]

**Pull request merged**